File: Bean.Client/app/plants/plant.ts
import { IBaseObject } from '../shared/base';
import { Status } from '../shared/enums';

interface IPlant {
    id: number;
    name: string;
    latinName: string;
    familyId: number;
    family: string;
    binder: string;
    plantingIn: Date;
    transplantOut: Date;
    directOut: Date;
    directGHSummer: Date;
    directGHWinter: Date;
    distanceBetweenPlants: number;
    distanceBetweenRows: number;
    yield: number;
    comment: string;
    coldHardy: boolean;
    quantityOnHand: number;
    companionPlants: Plant[],
    harmfulPlants: Plant[],
    status: Status;
}

export class Plant implements IPlant, IBaseObject {

    isNew = true;

    id = 0;
    name = "";
    latinName = "";
    familyId = 0;
    family = "";
    binder = "";
    plantingIn = new Date();
    transplantOut = new Date();
    directOut = new Date();
    directGHSummer = new Date();
    directGHWinter = new Date();
    distanceBetweenPlants = 0;
    distanceBetweenRows = 0;
    yield = 0;
    comment: string;
    coldHardy: boolean;
    quantityOnHand = 0;
    companionPlants = new Plant[0]();
    harmfulPlants = new Plant[0]();
    status = Status.Enabled;

    constructor() { }
}

File: Bean.API/Bean.POCO/Plant.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;

namespace Bean.POCO
{
    [DataContract(IsReference = true)]
    public class Plant : BasePOCO
    {
        [DataMember]
        [Key]
        public int Id { get; set; }

        [DataMember]
        public int FamilyId { get; set; }

        [DataMember]
        [Required(AllowEmptyStrings = false, ErrorMessage = "Family is required")]
        [ForeignKey("FamilyId")]
        public Family Family { get; set; }

        [DataMember]
        [Required(AllowEmptyStrings = false, ErrorMessage = "Name is required")]
        [MaxLength(50, ErrorMessage = "Name must be least then 50 characters")]
        [MinLength(3)]
        public string Name { get; set; }

        [DataMember]
        [MaxLength(50)]
        public string LatinName { get; set; }

        [DataMember]
        [MaxLength(50, ErrorMessage = "Species must be least then 50 characters")]
        public string Species { get; set; }

        [DataMember]
        public DateTime? PlantingIN { get; set; }

        [DataMember]
        public DateTime? TransplantOUT { get; set; }

        [DataMember]
        public DateTime? DirectOUT { get; set; }

        [DataMember]
        /// <summary>
        /// This date represent when you have to direct plant it into the green house at summer.
        /// </summary>
        public DateTime? DirectGHSummer { get; set; }

        [DataMember]
        /// <summary>
        /// This date represent when you have to direct plant it into the green house at winter.
        /// </summary>
        public DateTime? DirectGHWinter { get; set; }

        [DataMember]
        public int? DistanceBetweenPlants { get; set; }

        [DataMember]
        public int? DistanceBetweenRows { get; set; }

        [DataMember]
        public decimal? Yield { get; set; }

        [DataMember]
        public bool IsColdHardy { get; set; }

        [DataMember]
        public int QuantityOnHand { get; set; }

        [DataMember]
        [MaxLength(500)]
        public string Comment { get; set; }

        [DataMember]
        public List<Plant> CompanionPlants { get; set; }

        [DataMember]
        public List<Plant> HarmfulPlants { get; set; }

        [DataMember]
        [Required(ErrorMessage = "Status is required")]
        public Status Status { get; set; }

    }
}
